base/chrome_trace.cc

```diff
@@ -35,7 +36,7 @@ void chrome_trace::write_event(const char* name, const char* cat, char type) {
   // initializer.
   thread_local std::string pid_tid_str = []() {
     std::stringstream ss;
-    ss << "\",\"pid\":0,\"tid\":" << std::this_thread::get_id();
+    ss << "\",\"pid\":0,\"tid\":\"" << std::this_thread::get_id() << "\"";
```

Comment:
I've been meaning to change this anyways, because it causes the buffer to be flushed a lot more often (these ridiculous thread ID strings are a large fraction of the total size of an event).

How about this instead:
```
static std::atomic<int> next_thread_id = 0;
thread_local std::string pid_tid_str = []() {
    std::stringstream ss;
    ss << "\",\"pid\":0,\"tid\":" << next_thread_id++;
    return ss.str();
}();
```
`

Reply:
Thanks, that's a good idea.